sdk/cosmosdb/cosmos/review/cosmos.api.md

```diff
@@ -1715,6 +1717,12 @@ export const QueryMetricsConstants: {
     SchedulingMetricsText: string;
 };
 
+// @public (undocumented)
+export interface QueryOperationOptions {
+    [key: string]: any;
```

Comment:
Why do we need this indexer? This usually opens the door to user errors. For example if I had a typo building my QueryOperationOptions and created the object like this

```ts
const options: QueryOperationOptions = { rucapPerOperation: 123 } 
```

TypeScript will not complain because [key: string]: any is telling it to allow any property of any type. However the library won't know what to do with this and the ruCapPerOperation will not be sent. Diagnosing these errors is very time-consuming and hard to nail down.

I'd recommend not using the indexer unless there is a very strong justification for it.

Reply:
addressed